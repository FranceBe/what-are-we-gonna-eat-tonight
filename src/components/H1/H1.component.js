import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { fontSizes, spaces, colors } from 'styles/variables'

export const H1Default = styled.h1`
  font-size: ${fontSizes.px_30};
  text-align: center;
  text-shadow: ${spaces.xtraSmall} ${spaces.xtraSmall} ${colors.primary_default};
`

export class H1 extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any,
  }

  render() {
    return (
      <H1Default>{this.props.children}</H1Default>
    )
  }
}

export default H1
