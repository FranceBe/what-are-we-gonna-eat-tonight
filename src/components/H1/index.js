import H1Component, { H1, H1Default } from 'components/H1/H1.component'

export { H1, H1Default }

export default H1Component
