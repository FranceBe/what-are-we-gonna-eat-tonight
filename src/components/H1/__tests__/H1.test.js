import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import H1, { H1Default } from 'components/H1'

describe('H1', () => {
  describe('Styled component', () => {
    describe('H1Default', () => {
      it('should match snapshot', () => {
        const H1DefaultTree = renderer.create(<H1Default/>).toJSON()
        expect(H1DefaultTree).toMatchSnapshot()
      })
      it('should have font-size set to 1.875rem', () => {
        const H1DefaultTree = renderer.create(<H1Default/>).toJSON()
        expect(H1DefaultTree).toHaveStyleRule('font-size', '1.875rem')
      })
      it('should have text-align set to center', () => {
        const H1DefaultTree = renderer.create(<H1Default/>).toJSON()
        expect(H1DefaultTree).toHaveStyleRule('text-align', 'center')
      })
      it('should have text-shadow set to 1px 1px #EBA789', () => {
        const H1DefaultTree = renderer.create(<H1Default/>).toJSON()
        expect(H1DefaultTree).toHaveStyleRule('text-shadow', '1px 1px #EBA789')
      })
    })
  })
  describe('rendering', () => {
    describe('default props', () => {
      const children = 'I am an H1 title :)'
      it('should render H1Default component with its children', () => {
        const H1Wrapper = shallow(<H1>{children}</H1>)
        expect(H1Wrapper.find(H1Default)).toHaveLength(1)
        expect(H1Wrapper.find(H1Default).text()).toBe('I am an H1 title :)')
      })
    })
  })
})
