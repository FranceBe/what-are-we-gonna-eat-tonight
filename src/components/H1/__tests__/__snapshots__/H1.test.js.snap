// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Styled component H1Default should match snapshot 1`] = `
.c0 {
  font-size: 1.875rem;
  text-align: center;
  text-shadow: 1px 1px #EBA789;
}

<h1
  className="c0"
/>
`;
